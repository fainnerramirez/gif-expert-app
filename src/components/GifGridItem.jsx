import React from "react";
import PropsTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="col animate__animated animate__fadeInUp">
      <div className="card" style={{ width: "17rem" }}>
        <img src={url} className="card-img-top" alt={title} />
        <div className="card-body bg-warning">
          <h5 className="card-title text-center">{title}</h5>
        </div>
      </div>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropsTypes.string.isRequired,
  url: PropsTypes.string.isRequired,
};

export default GifGridItem;
