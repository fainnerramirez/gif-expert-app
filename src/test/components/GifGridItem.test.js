import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en el archivo gifGridItem.js", () => {
  const title = "Title example";
  const url = "http://localhost/image.png";
  const view = shallow(<GifGridItem title={title} url={url} />);

  test("Se debe mostrar el componente correctamente", () => {
    expect(view).toMatchSnapshot();
  });

  test("El componente debe tener los atributos obligatorios", () => {
    expect(view).toMatchSnapshot();
  });

  test("debe de tener un pàrrafo con el tìtulo", () => {
    const parrafoTitulo = view.find("h5");
    const text = parrafoTitulo.text().trim();

    expect(text).toBe(title);
  });

  test("Debe de mostrar la url y el alt props de la etiqueta img", () => {
    const img = view.find("img");

    const alt = img.props().alt;
    const src = img.props().src;

    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("Debe de tener la clase animate__fadeInUp", () => {
    const div = view.find("div").at(0);

    const className = div.props().className;

    expect(className.includes("animate__fadeInUp")).toBe(true);
  });
});
