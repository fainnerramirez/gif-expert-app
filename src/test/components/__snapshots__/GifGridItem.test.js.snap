// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo gifGridItem.js El componente debe tener los atributos obligatorios 1`] = `
<div
  className="col animate__animated animate__fadeInUp"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      alt="Title example"
      className="card-img-top"
      src="http://localhost/image.png"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Title example
      </h5>
    </div>
  </div>
</div>
`;

exports[`Pruebas en el archivo gifGridItem.js Se debe mostrar el componente correctamente 1`] = `
<div
  className="col animate__animated animate__fadeInUp"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      alt="Title example"
      className="card-img-top"
      src="http://localhost/image.png"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Title example
      </h5>
    </div>
  </div>
</div>
`;
